cvat-sdk/cvat_sdk/auto_annotation/interface.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2023 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import List, Sequence
+
+import attrs
+import PIL.Image
+from typing_extensions import Protocol
+
+import cvat_sdk.models as models
+
+
+@attrs.frozen(kw_only=True)
+class DetectionFunctionSpec:
```

Comment:
From the user's perspective, I would not recommend using PatchedLabelRequest and LabeledShapeRequest. They are counterintuitive. 

I will recommend defining alternative naming. Something like Label or Shape.

Another approach is to define some high-level interfaces like LabelingSpec and LabeledFrame and use them.

In both cases, the interface will be clear and attractive.

Also, even I can read comments about DetectionFunctionContext, I'm not sure that we need it. How are you going to use it in the future?



defining the "Label" interface in HL SDK. Something like Sequence[Label] would be better (PatchedLabelRequest -> Label). Another approach is to provide LabelingSpec in SDK and define it as a sequence of labels.

The same comment I can leave about LabeledShapeRequest. It is better to have just Shape. Let's try to improve our high-level interface for users. We can probably define LabeledFrame interface or something like that

Reply:
I think I'll do something of a middle ground. I'll add factory functions with nice names and nice interfaces that construct the same old `*Request` objects. That way, the function implementation can look nice and clean, but we're not introducing any new concepts (and the users can still construct the low-level request objects if the factory functions are insufficient for some reason).

> Also, even I can read comments about DetectionFunctionContext, I'm not sure that we need it. How are you going to use it in the future?

See the thread directly above this one.

Reply:
> I'll add factory functions with nice names and nice interfaces

I did it.